WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/AuthorSelectionAdapter.kt

```diff
@@ -0,0 +1,137 @@
+package org.wordpress.android.ui.posts.adapters
+
+import android.content.Context
+import android.database.DataSetObserver
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.AppCompatImageView
+import android.support.v7.widget.AppCompatTextView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.SpinnerAdapter
+import androidx.annotation.CallSuper
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.ui.posts.PostListMainViewModel.AuthorFilterListItemUIState
+import org.wordpress.android.ui.utils.UiHelpers
+import org.wordpress.android.util.GravatarUtils
+import org.wordpress.android.util.image.ImageManager
+import java.lang.ref.WeakReference
+import javax.inject.Inject
+
+class AuthorSelectionAdapter(context: Context) : SpinnerAdapter {
+    @Inject lateinit var imageManager: ImageManager
+    @Inject lateinit var uiHelpers: UiHelpers
+
+    private var observers: MutableList<WeakReference<DataSetObserver>> = mutableListOf()
+    private val items = mutableListOf<AuthorFilterListItemUIState>()
+
+    init {
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
+    override fun getCount(): Int = items.count()
+
+    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: DropdownViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection_dropdown, parent, false)
+            holder = DropdownViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as DropdownViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getItem(position: Int): AuthorFilterListItemUIState = items[position]
+
+    override fun getItemId(position: Int): Long = position.toLong()
+
+    override fun getItemViewType(position: Int): Int = 0
+
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        var view: View? = convertView
+        val holder: NormalViewHolder
+
+        if (view == null) {
+            val inflater = LayoutInflater.from(parent.context)
+            view = inflater.inflate(R.layout.post_list_author_selection, parent, false)
+            holder = NormalViewHolder(view)
+            view.tag = holder
+        } else {
+            holder = view.tag as NormalViewHolder
+        }
+
+        holder.bind(items[position], imageManager, uiHelpers)
+
+        return view!!
+    }
+
+    override fun getViewTypeCount(): Int = 1
+    override fun hasStableIds(): Boolean = false
```

Comment:
Nitpick: This will not make any difference, but when we use the proper ids for the author selection, we can return `true` here. I know it's only used for performance optimization and there is nothing to optimize here, but it might throw off another developer.

Reply:
I would of course change this back when providing a stable id.

Reply:
This was resolved when the other comment was resolved in 0bbec1c54fae6b5923f6301670c777582947a165